pkg/prometheus/promcfg.go

```diff
@@ -4496,17 +4496,25 @@ func (cg *ConfigGenerator) appendOTLPConfig(cfg yaml.MapSlice) (yaml.MapSlice, e
 		return cfg, fmt.Errorf("OTLP configuration is only supported from Prometheus version 2.55.0")
 	}
 
-	return append(
-		cfg,
-		yaml.MapItem{
-			Key: "otlp",
-			Value: yaml.MapSlice{
-				{
-					Key:   "promote_resource_attributes",
-					Value: otlpConfig.PromoteResourceAttributes,
-				},
-			},
-		}), nil
+	otlp := yaml.MapSlice{}
+
+	if len(otlpConfig.PromoteResourceAttributes) > 0 {
+		otlp = cg.WithMinimumVersion("2.55.0").AppendMapItem(otlp,
+			"promote_resource_attributes",
+			otlpConfig.PromoteResourceAttributes)
+	}
+
+	if otlpConfig.TranslationStrategy != nil {
+		otlp = cg.WithMinimumVersion("3.0.0").AppendMapItem(otlp,
+			"translation_strategy",
+			otlpConfig.TranslationStrategy)
+	}
+
+	if len(otlp) == 0 {
+		return cfg, nil
+	}
+
+	return cg.WithMinimumVersion("2.55.0").AppendMapItem(cfg, "otlp", otlp), nil
```

Comment:
(nit) it's already tested above twice because we return if version is < 2.55 and if otlp is empty.
```suggestion
	return cg.AppendMapItem(cfg, "otlp", otlp), nil
```